.dotnet/src/Custom/Chat/StreamingChatCompletionUpdate.cs

```diff
@@ -99,6 +108,11 @@ public partial class StreamingChatCompletionUpdate
     public StreamingChatFunctionCallUpdate FunctionCallUpdate => (Choices.Count > 0)
         ? Choices[0].Delta.FunctionCall
         : null;
+    
+    // CUSTOM: Flattened choice delta property.
+    public string RefusalMessage => (Choices.Count > 0)
```

Comment:
Just double-checking: The refusal is sent as a whole in the streaming case? Or is it sent token by token? If sent token by token, we should call it "RefusalUpdate".

Reply:
Need to exercise to see. It doesn't appear to be adequately documented anywhere.

Reply:
I just pushed an update that explicitly tests the population of `refusal` for both streaming and non-streaming.

Interestingly, despite the spec, I can't get `refusal` to populate when streaming. Using the same input as the non-streaming case, I see `refusal` emitted on a single chunk as "null" and then the refusal in `content`, e.g. (first three chunks' choices arrays):

```
"choices":[{"index":0,"delta":{"role":"assistant","content":"","refusal":null},"logprobs":null,"finish_reason":null}]
"choices":[{"index":0,"delta":{"content":"I'm"},"logprobs":null,"finish_reason":null}]
"choices":[{"index":0,"delta":{"content":" sorry"},"logprobs":null,"finish_reason":null}]
```

For now, I'm going to proceed with the speculation that:
- `refusal` not having parity behavior when streaming is a bug
- the single instance observed of `refusal` implies that, like `role` being set, when the above bug is fixed it'll be a "set" and not a "patch" -- and hence that `Refusal` is appropriate vs. `RefusalUpdate`.

Reply:
I looked at the Python and JavaScript libraries, and they have a "RefusalDeltaEvent":
🔗 https://github.com/openai/openai-python/blob/main/helpers.md#refusaldeltaevent
🔗 https://github.com/openai/openai-node/blob/master/helpers.md#onrefusaldelta-props-refusaldeltaevent--

Reply:
Ah, good find! That's enough to convince me: renaming the streaming properties to `RefusalUpdate`.